pyfar/dsp/fractional_octave_smoothing.py

```diff
@@ -0,0 +1,317 @@
+    # Push front for lower limit
+    lower_limit = np.append(low, base)
+    # Pad arrays front:
+    pad_front = int(upper_limit[0])
+    upper_limit = np.pad(upper_limit, (pad_front, 0))
+    lower_limit = np.pad(lower_limit, (pad_front, 0))
+    # Pad array back to 'size' (length of largest smoothing window at highest
+    # frequency bin k)
+    upper_limit = np.pad(upper_limit, (0, size-len(upper_limit)))
+    lower_limit = np.pad(lower_limit, (0, size-len(lower_limit)))
+    # Return upper and lower limits of k-th freq bin in one array:
+    return np.concatenate((
+        upper_limit.reshape(1, -1),
+        lower_limit.reshape(1, -1)))
+
+
+def data_padder(data, pad_width, mean_size):
+    """
+    Pads data array of shape (N, M) to data matrix of shape
+    (N, M, M+pad_width). The output data contains M copies of the input data
+    rows padded by 'pad_width' with the mean value of the last data samples.
+    The number of samples used to compute the mean for each row is specified in
+    the array 'mean_size'.
+
+    Parameters
+    ----------
+    data : ndarray
+        Array of shape (N, M) of the data.
+        (N= Number of channels, M=Length of Signal)
+    pad_width : int
+        Padding length
+    mean_size : ndarray
+        Array with numbers of samples used to compute the mean.
+
+    Returns
+    -------
+    ndarray
+        Padded data matrix containing the data values and specified padding
+        value for each frequency bin.
+
+    Raises
+    ------
+    ValueError
+        Data array must be 2D.
+    """
+    if not data.ndim == 2:
+        raise ValueError('Data array must be 2D.')
+    padded = np.array([np.pad(data, ((0, 0), (0, pad_width)),
+                              'mean', stat_length=(m)) for m in mean_size])
+    # move channel axis to front and return
+    return np.moveaxis(padded, 1, 0)
+
+
+def polar2cartesian(amplitude, phase):
+    """
+    Converts two arrays of amplitude and phase into one array
+    of complex numbers in cartesian form.
+
+    Parameters
+    ----------
+    amplitude : ndarray
+        Array of amplitude values.
+    phase : ndarray
+        Array of phase values.
+
+    Returns
+    -------
+    ndarray
+        Array of same shape as input arrays with complex numbers.
+
+    Raises
+    ------
+    ValueError
+        Arrays must have same shape.
+    """
+    if not (amplitude.shape == phase.shape):
+        raise ValueError("Arrays must have same shapes.")
+    # Save input shape
+    input_shape = amplitude.shape
+    # Reshape for cmath rect to 1D arrays:
+    reshaped_amplitude = amplitude.reshape(1, -1)[0]
+    reshaped_phase = phase.reshape(1, -1)[0]
+    # Apply cmath rect on elements:
+    cartesian = np.array([cm.rect(a, p) for a, p in
+                          zip(reshaped_amplitude, reshaped_phase)])
+    return cartesian.reshape(input_shape)
+
+
+def frac_smooth_signal(signal, smoothing_width):
+    """
+    Method to smooth a given signal.
```

Comment:
If possible, docstrings should start with a single line, e.g.:
```suggestion
    Smooth spectrum of a signal with fractional octave width.
```

Reply:
Done.